controllers/front/SmailyCartCron.php

```diff
@@ -45,107 +45,157 @@ public function init()
-                            }
-                        }
-                        $count++;
-                    }
-                    // Smaily api query.
-                    $query = array(
-                        'autoresponder' => $autoresponder_id,
-                        'addresses' => array($adresses)
-                    );
-                    // Send cart data to smaily api.
-                    $response = $this->module->callApi('autoresponder', $query, 'POST');
-                    // If email sent successfully update sent status in database.
-                    if (array_key_exists('success', $response) &&
-                        isset($response['result']['code']) &&
-                        $response['result']['code'] === 101) {
-                            $this->updateSentStatus($id_customer, $id_cart);
-                    } else {
-                        $this->module->logTofile('smaily-cart.txt', Tools::jsonEncode($response));
+        // Settings.
+        $autoresponder = unserialize(Configuration::get('SMAILY_CART_AUTORESPONDER'));
+        $autoresponder_id = pSQL($autoresponder['id']);
+        $delay = pSQL(Configuration::get('SMAILY_ABANDONED_CART_TIME'));
+        $sync_fields = unserialize(Configuration::get('SMAILY_CART_SYNCRONIZE_ADDITIONAL'));
+
+        $abandoned_carts = $this->getAbandonedCarts();
+        foreach ($abandoned_carts as $abandoned_cart) {
+            $cart_updated_time = strtotime($abandoned_cart['date_upd']);
+            $reminder_time = strtotime('+' . $delay . ' minutes', $cart_updated_time);
+            $current_time = strtotime(date('Y-m-d H:i') . ':00');
+            // Don't continue if cart delay time has not passed.
+            if ($current_time < $reminder_time) {
+                continue;
+            }
+
+            // Get cart by id.
+            $id_customer = (int) $abandoned_cart['id_customer'];
+            $id_cart = (int) $abandoned_cart['id_cart'];
+            $cart = new Cart($abandoned_cart['id_cart']);
+
+            // Get cart products.
+            $products = $cart->getProducts();
+            // Don't continue if no products in cart.
+            if (empty($products)) {
+                continue;
+            }
+
+            // Initialize Smaily query.
+            $adresses = array(
+                'email' => $abandoned_cart['email'],
+            );
+
+            if (in_array('first_name', $sync_fields)) {
+                $adresses['first_name'] = $abandoned_cart['firstname'];
+            }
+
+            if (in_array('last_name', $sync_fields)) {
+                $adresses['last_name'] = $abandoned_cart['lastname'];
+            }
+
+            // Populate abandoned cart with empty values for legacy api.
+            $fields_available = array(
+                'name',
+                'description',
+                'price',
+                'category',
```

Comment:
The product category is not a field that is used by other plugins. Should I remove it from Prestashop also?

Reply:
If it is not used by other plugins, then I suggest we remove it.

Reply:
Removed in https://github.com/sendsmaily/smaily-prestashop-module/pull/12/commits/2a56e8af7442151d91968af40dbc202f4bc9a7ae